src/tests/ftest/tags.py

```diff
@@ -144,14 +161,17 @@ def is_test_subset(self, tags1, tags2):
         """
         tests1 = set(self.__tags_to_tests(tags1))
         tests2 = set(self.__tags_to_tests(tags2))
-        return tests1 and tests2 and tests1.issubset(tests2)
+        return bool(tests1) and bool(tests2) and tests1.issubset(tests2)
 
     def __tags_to_tests(self, tags):
         """Convert a list of tags to the tests they would run.
 
         Args:
             tags (list): list of sets of tags
         """
+        # Convert to TagSet to handle negative matching
+        for idx, _tags in enumerate(tags):
+            tags[idx] = TagSet(_tags)
```

Comment:
I considered using TagSet in the class itself so it's used everywhere, but we only really need negative matching for the user facing tools. And it does add a slight noticeable overhead to set operations, so I didn't want to slow down the core code